### PR TITLE
Soft Delete Recommendation

### PR DESCRIPTION
**Pull request by alisakina99**
# Overview/Summary

added recommendation for enabling soft delete including an arg query

Screenshot 2024-04-22 141920

## Related Issues/Work Items

AB#29823

## This PR fixes/adds/changes/removes

1. changes azure-resources\RecoveryServices\vaults\recommendations.yaml
2. adds azure-resources\RecoveryServices\vaults\kql\9e39919b-78af-4a0b-b70f-c548dae97c25

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/RecoveryServices/vaults/recommendations.yaml [hunk 1]
```diff
@@ -81,3 +81,31 @@
...
       url: "https://learn.microsoft.com/azure/backup/backup-azure-arm-vms-prepare"
+
+- description: Enable Soft Delete for Recovery Services Vaults in Azure Backup
+  aprlGuid: 9e39919b-78af-4a0b-b70f-c548dae97c25
+  recommendationTypeId: null
+  recommendationControl: Disaster Recovery
+  recommendationImpact: Medium
+  recommendationResourceType: Microsoft.RecoveryServices/vaults
+  recommendationMetadataState: Active
+  longDescription: |
```
Comment (ejhenry) [inline azure-resources/RecoveryServices/vaults/recommendations.yaml:92]: Please reword the longDescription to be 300 characters or less, including spaces.

File: azure-resources/RecoveryServices/vaults/recommendations.yaml [hunk 2]
```diff
@@ -81,3 +81,31 @@
...
+    With soft delete, even if a malicious actor deletes a backup (or backup data is accidentally deleted), the backup data is retained for 14 additional days, allowing the recovery of that backup item with no data loss. The additional 14 days of retention for backup data in the "soft delete" state don't incur any cost to you.
+    Soft delete is enabled by default on newly created vaults to protect backup data from accidental or malicious deletes. Disabling this feature isn't recommended. The only circumstance where you should consider disabling soft delete is if you're planning on moving your protected items to a new vault, and can't wait the 14 days required before deleting and reprotecting (such as in a test environment).
+    Soft delete protection is available for certain services such as Azure VMs.  potentialBenefits: Enhances disaster recovery capabilities
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
```
Comment (ejhenry) [inline azure-resources/RecoveryServices/vaults/recommendations.yaml:101]: Please include only a single learn more link/

**Review by ejhenry — COMMENTED** [review]
Thanks @alisakina99! The isSoftDelete column in the query output needs to be renamed to param1 to conform to the expected ARG query results.

**Review by ejhenry — APPROVED** [review]
LGTM 👍

**Review by Dost2010 — APPROVED** [review]

**Pull request merged**